Library/Homebrew/utils/spdx.rb

```diff
@@ -8,32 +8,51 @@
 module SPDX
   module_function
 
-  DATA_PATH = (HOMEBREW_DATA_PATH/"spdx").freeze
+  DATA_PATH = T.let((HOMEBREW_DATA_PATH/"spdx").freeze, Pathname)
   API_URL = "https://api.github.com/repos/spdx/license-list-data/releases/latest"
   LICENSEREF_PREFIX = "LicenseRef-Homebrew-"
   ALLOWED_LICENSE_SYMBOLS = [
     :public_domain,
     :cannot_represent,
   ].freeze
 
+  sig { returns(T::Hash[String, T.untyped]) }
   def license_data
-    @license_data ||= JSON.parse (DATA_PATH/"spdx_licenses.json").read
+    @license_data ||= T.let(JSON.parse((DATA_PATH/"spdx_licenses.json").read), T.nilable(T::Hash[String, T.untyped]))
   end
 
+  sig { returns(T::Hash[String, T.untyped]) }
   def exception_data
-    @exception_data ||= JSON.parse (DATA_PATH/"spdx_exceptions.json").read
+    @exception_data ||= T.let(JSON.parse((DATA_PATH/"spdx_exceptions.json").read),
+                              T.nilable(T::Hash[String, T.untyped]))
   end
 
+  sig { returns(String) }
   def latest_tag
-    @latest_tag ||= GitHub::API.open_rest(API_URL)["tag_name"]
+    @latest_tag ||= T.let(GitHub::API.open_rest(API_URL)["tag_name"], T.nilable(String))
   end
 
+  sig { params(to: Pathname).void }
   def download_latest_license_data!(to: DATA_PATH)
     data_url = "https://raw.githubusercontent.com/spdx/license-list-data/#{latest_tag}/json/"
     Utils::Curl.curl_download("#{data_url}licenses.json", to: to/"spdx_licenses.json")
     Utils::Curl.curl_download("#{data_url}exceptions.json", to: to/"spdx_exceptions.json")
   end
 
+  sig {
+    params(
+      license_expression: T.any(
+        String,
+        Symbol,
+        T::Hash[T.any(Symbol, String), T.untyped],
+        T::Array[String],
+      ),
+    ).returns(
+      [
+        T::Array[T.any(String, Symbol)], T::Array[String]
```

Comment:
Would be nice if this could use either `String` or `Symbol` consistently as the Hash keys/first Array values? Maybe outside the scope of this PR.